File: Lakshay Garg/Leetcode/Sorting/Maximum Area of a Piece of Cake After Horizontal and Vertical Cuts.cpp
class Solution {
public: 
    const int MOD=1e9+7;
    int maxArea(int h, int w, vector<int>& hc, vector<int>& vc) {
        hc.push_back(h);
        vc.push_back(w);
        sort(hc.begin(),hc.end());
        sort(vc.begin(),vc.end());
        int x=hc[0],y=vc[0];
        for(int i=1;i<hc.size();i++) x=max((hc[i]-hc[i-1]),x);
        for(int i=1;i<vc.size();i++) y=max((vc[i]-vc[i-1]),y);
        long long ans=(1LL*x*y)%MOD;
        return (int)ans;
    }
};

File: Anya_Seth/Sorting/Maximum Area of a Piece of Cake After Horizontal and Vertical Cuts.java
class Solution {
    public int maxArea(int h, int w, int[] horizontalCuts, int[] verticalCuts){
        int MODULO = (int) 1e9 + 7;
        Arrays.sort(horizontalCuts);
        Arrays.sort(verticalCuts);
        long maxH = horizontalCuts[0];
        long maxV = verticalCuts[0];

        for(int i = 1 ; i < horizontalCuts.length ; i++){
            maxH = Math.max(maxH, horizontalCuts[i] - horizontalCuts[i - 1]);
        }
        maxH = Math.max(maxH, h - horizontalCuts[horizontalCuts.length - 1]);

        for(int i = 1 ; i < verticalCuts.length ; i++){
            maxV = Math.max(maxV, verticalCuts[i] - verticalCuts[i - 1]);
        }
        maxV = Math.max(maxV, w - verticalCuts[verticalCuts.length - 1]);

        long maxArea = (maxH * maxV) % MODULO;
        return (int) maxArea;
    }
}
